src/codegen/sys/cargo_toml.rs

```diff
@@ -117,10 +117,10 @@ fn fill_in(root: &mut Table, env: &Env) {
         let lib_name = ns.package_name.as_ref().unwrap();
 
         let meta = upsert_table(meta, nameutil::lib_name_to_toml(lib_name));
-        if !meta.contains_key("name") {
-            set_string(meta, "name", lib_name);
-        }
-        set_string(meta, "version", env.config.min_cfg_version.to_string());
+        meta.entry("name")
+            .or_insert_with(|| Value::String(lib_name.to_owned()));
```

Comment:
Doesn't this basically revert https://github.com/gtk-rs/gir/pull/1022 ?

Reply:
This time I fixed my mistake from #1011. Since that PR didn't bother to state an actual bug has been fixed _nor how_, I couldn't be bothered to explain either.

Hint: the difference is already in the signature for `fn entry` vs `fn get_mut`.

Reply:
Please add a code comment explaining all this so that we know exactly what is going on in case we need to update it once again in the future.

Reply:
@GuillaumeGomez Added an extra comment explaining this allows overriding `name` and `version` by hand. More explanation and links are in the commit, unless you want those inline as well.

I doubt you intend to explain the difference between `.get_mut` and `.entry` in an inline comment though, that is better suited for the commit addressing the bug itself.

Reply:
> Since that PR didn't bother to state an actual bug has been fixed nor how, I couldn't be bothered to explain either.

You mean #1022? The full explanation is in the linked issue #1020 :)

Reply:
Not that I can see, and even if it did, such an explanation does not show up in `git blame` nor `git show`.

Reply:
As #1020 explains, autogenerating the -sys crate put a dependency with a wrong pkg-config name into `Cargo.toml`: `tracker-sparql-3.0` it should've been, `tracker_sparql_3.0` it was. That's what #1022 fixed.

Reply:
@sdroege Thanks for providing a perfect example to the problem (lacking commit titles/messages) I'm highlighting here.

My mistake in #1011 causes no `name = "name-with-hyphens"` override to show up _at all_, because `get_mut` does not allow modifying the underlying map with insertions (it returns `Option<&mut ItemType>`, so only a mutable reference to the item _if it exists_). That's what `fn entry` is for.
Note that `set_string` - reintroduced by #1022 - does not perform this conversion - `lib_name` is already "canonicalized" with hyphens.

I think the author is seeing underscores because it falls back to `[lib] name = "name_with_underscores"`, again because the unversioned `[...system-deps...]` table is generated without `name` entry.